src/models/event.js

```diff
@@ -712,6 +713,13 @@ utils.extend(module.exports.MatrixEvent.prototype, {
 
         this.emit("Event.beforeRedaction", this, redaction_event);
 
+        // before removing content, preserve the rel_type
+        // if this is a relation, so we can still tell the
+        // difference between a redacted edit and message
+        const relation = this.getRelation();
+        if (relation) {
+            this._relationTypeBeforeRedaction = relation.rel_type;
```

Comment:
is this field persisted? Concern would be it showing correctly at the time it happened, but when switching rooms or refreshing the page the double redaction comes back.

Reply:
Don't think it needs to be. All events are kept in memory, so switching rooms shouldn't clear this.
As for refreshing the page, both the edit and its redaction should be loaded in the same order when processing the sync accumulator blob, so that should also work.

Reply:
Argh, you might be right. Just saw them popup again (fetched through /messages I presume). I was assuming redacted messages wouldn't show up through /messages, but looks like that is not the case...